syz-manager/rpc.go

```diff
@@ -83,11 +85,44 @@ func startRPCServer(mgr *Manager) (*RPCServer, error) {
 	return serv, nil
 }
 
+var (
+	once                sync.Once
+	coverFilterFilename string
+	enabledCoverFilter  bool
+)
+
+func (serv *RPCServer) update(a *rpctype.ConnectArgs, r *rpctype.ConnectRes) error {
+	var (
+		err         error
+		coverFilter map[uint32]uint32
+	)
+	once.Do(func() {
+		err = serv.mgr.updateKernelModules(a)
+		if err != nil {
+			return
+		}
+		coverFilterFilename, coverFilter, enabledCoverFilter, err = serv.mgr.updateCoverFilter()
+	})
+	if err != nil {
+		return err
+	}
+	if coverFilterFilename != "" {
+		data, err := ioutil.ReadFile(coverFilterFilename)
+		if err != nil {
+			return err
+		}
+		r.CoverFilterBitmap = data
+	}
+	r.EnabledCoverFilter = enabledCoverFilter
+	serv.coverFilter = coverFilter
+	return nil
+}
+
 func (serv *RPCServer) Connect(a *rpctype.ConnectArgs, r *rpctype.ConnectRes) error {
 	log.Logf(1, "fuzzer %v connected", a.Name)
 	serv.stats.vmRestarts.inc()
 
-	corpus, bugFrames, enabledCoverFilter := serv.mgr.fuzzerConnect()
+	corpus, bugFrames := serv.mgr.fuzzerConnect()
```

Comment:
We now do 3 callbacks to serv.mgr during connect, it's reasonable to combine all 3 into the existing mgr.fuzzerConnect callback. Pass a.Modules to mgr.fuzzerConnect and make it return coverFilter.